x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/cohere/embeddings/CohereEmbeddingsTaskSettings.java

```diff
@@ -106,10 +134,44 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalEnum(truncation);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        CohereEmbeddingsTaskSettings that = (CohereEmbeddingsTaskSettings) o;
+        return inputType == that.inputType && truncation == that.truncation;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(inputType, truncation);
+    }
+
     public CohereEmbeddingsTaskSettings overrideWith(CohereEmbeddingsTaskSettings requestTaskSettings) {
-        var inputTypeToUse = requestTaskSettings.inputType() == null ? inputType : requestTaskSettings.inputType();
-        var truncationToUse = requestTaskSettings.truncation() == null ? truncation : requestTaskSettings.truncation();
+        if (requestTaskSettings.equals(EMPTY_SETTINGS)) {
+            return this;
+        }
+
+        var inputTypeToUse = requestTaskSettings.getInputType() == null ? inputType : requestTaskSettings.getInputType();
+        var truncationToUse = requestTaskSettings.getTruncation() == null ? truncation : requestTaskSettings.getTruncation();
 
         return new CohereEmbeddingsTaskSettings(inputTypeToUse, truncationToUse);
     }
```

Comment:
Is this "overrideWith" method an instance of some type of common pattern? I've never seen this before and I find its behaviour with regards to object referencing to be unintuitive. Its weird to me that an object would have a method which takes an instance of the same class and creates a whole new instance of the same class with the copied attributes. 

I suppose this is required because the attributes of CohereEmbeddingsTaskSettings are final and so this class is effectively immutable, but then, if we already have an instance of CohereEmbeddingsTaskSettings with the required attributes, why do we need to create a copy of it?

Overall, perhaps this function can just be renamed to `copy` or override `clone`, or create a copy-constructor?

Reply:
Fair points, I'm going to combine the `set` and `overrideWith` into a single static method `of`. I think a static method might be a better fit than a copy constructor because of the logic around figuring out which field to use from each of the objects passed in. Copy constructors seems like a better if we were only passing in a single `CohereEmbeddingsTaskSettings` object to simply copy every field.